cpp/arcticdb/async/async_store.hpp

```diff
@@ -280,36 +280,35 @@ class AsyncStore : public Store {
         return res;
     }
 
-    std::vector<VariantKey> batch_read_compressed(
+    folly::Future<std::vector<VariantKey>> batch_read_compressed(
         std::vector<entity::VariantKey> &&keys,
         std::vector<ReadContinuation> &&continuations,
         const BatchReadArgs & args) override {
         util::check(keys.size() == continuations.size(),
                     "keys and continuations must has then same number of elements");
-        std::vector<folly::Future<VariantKey>> batch;
-        batch.reserve(args.batch_size_);
-        std::vector<VariantKey> res;
-        res.reserve(keys.size());
-        for (std::size_t i = 0; i < keys.size(); ++i) {
-            auto &key = keys[i];
-            auto &cont = continuations[i];
-            batch.push_back(
-                async::submit_io_task(ReadCompressedTask(key, library_, storage::ReadKeyOpts{}))
-                    .via(&async::cpu_executor())
-                    .thenValue(SegmentFunctionTask{std::move(cont)}));
-
-            if(batch.size() == args.batch_size_) {
-                auto vec = folly::collect(batch).get();
-                res.insert(end(res), std::make_move_iterator(std::begin(vec)), std::make_move_iterator(std::end(vec)));
-                batch.clear();
-            }
-        }
-
-        if(!batch.empty()) {
-            auto vec = folly::collect(batch).get();
-            res.insert(end(res), std::make_move_iterator(std::begin(vec)), std::make_move_iterator(std::end(vec)));
+        
+        
+        std::vector<std::tuple<entity::VariantKey, int>> keys_and_continuations_idx(keys.size());
```

Comment:
Might want to use a `pair` instead. `first` and `second` is more readable than `std::get<>` IMO

Reply:
removed all this part